packages/actor-rdf-join-multi-sequential/lib/ActorRdfJoinMultiSequential.ts

```diff
@@ -0,0 +1,39 @@
+import {IActorQueryOperationOutput, IActorQueryOperationOutputBindings} from "@comunica/bus-query-operation";
+import {ActorRdfJoin, IActionRdfJoin} from "@comunica/bus-rdf-join";
+import {IActorArgs, IActorTest, Mediator} from "@comunica/core";
+import {IMediatorTypeIterations} from "@comunica/mediatortype-iterations";
+
+/**
+ * A comunica Multi Sequential RDF Join Actor.
+ * It accepts 3 or streams at once, joins the first two, and joins the result with the remaining streams.
+ */
+export class ActorRdfJoinMultiSequential extends ActorRdfJoin {
+
+  public readonly mediatorJoin: Mediator<ActorRdfJoin,
+    IActionRdfJoin, IMediatorTypeIterations, IActorQueryOperationOutput>;
+
+  constructor(args: IActorRdfJoinMultiSequentialArgs) {
+    super(args, 3, true);
+  }
+
+  protected async getOutput(action: IActionRdfJoin): Promise<IActorQueryOperationOutputBindings> {
+    // Join the two first streams, and then join the result with the remaining streams
+    const firstEntry: IActorQueryOperationOutputBindings = <IActorQueryOperationOutputBindings> await
+      this.mediatorJoin.mediate({ entries: [ action.entries[0], action.entries[1] ] });
+    const remainingEntries: IActorQueryOperationOutputBindings[] = action.entries.slice(2);
+    remainingEntries.unshift(firstEntry);
```

Comment:
could also do a slice 1 and then replace the first element with the new entry to do less array operations

Reply:
Ah yes, good catch.